src/components/VLink.vue

```diff
@@ -33,9 +33,11 @@ export default defineComponent({
   },
   setup(props) {
     const { app } = useContext()
-    const hasHref = computed(
-      () => typeof props.href === 'string' && !['', '#'].includes(props.href)
-    )
+    function checkHref(p: typeof props): p is { href: string } {
+      return typeof p.href === 'string' && !['', '#'].includes(p.href)
+    }
```

Comment:
This looks curious to me 😯 Is `p is { href: string }` the returning type or an extension of the params type?

Reply:
That makes sense for the part of the code using the `checkHref` function but my doubt is actually with the definition highlighted here 😅

Reply:
I see, sorry.

The `p is { href: string }` is the return type yes. Any return type with the `is` form is taken to be a predicate and the return value is interpreted as a `boolean`. It refines the type whatever the caller passes for `p` at the call site.

Reply:
Thank you for the explanation!

Reply:
Now I see! Thank you for this explanation, the example in the link made it crystal clear. The way Typescript works is interesting.